sirepo/template/elegant_lattice_importer.py

```diff
@@ -33,34 +34,43 @@
 _ELEGANT_TYPES = set(n for n in _SCHEMA.model if _ELEGANT_TYPE_RE.search(n))
 
 
-class ElegantRPNEval(object):
-
-    def eval_var(self, expr, depends, variables):
-        #TODO(robnagler) scan variable values for strings. Need to be parsable
-        var_list = ' '.join(map(lambda x: '{} sto {}'.format(variables[x], x), depends))
-        #TODO(pjm): security - need to scrub field value
-        #     execn  send top of string stack to UNIX and put result on numerical stack
-        #     execs     send top of string stack to UNIX and put output on string stack
-        # csh                       start and enter C shell subprocess
-        # cshs                       send top of string stack to C shell
-        # gets                       get string from input file
-        # seems like this would be bad, because you could construct a string that could be executed
-        # mudf    make user defined function from string stack (name commands mudf)
-        # open                       open input/output file
-        # puts                       put string to file
-        # sleep                       sleep for number of seconds
-        # @                       push command input file
-        pkdc('rpn variables={} expr="{}"', var_list, expr)
-        out = elegant_common.subprocess_output(['rpnl', '{} {}'.format(var_list, expr)])
-        if out is None:
-            return None, 'invalid'
-        if out:
-            return float(out.strip()), None
-        return None, 'empty'
-
-
 def elegant_code_var(variables):
-    return code_variable.CodeVar(variables, ElegantRPNEval())
+    _PI = 4 * math.atan(1)
+
+    class _P(code_variable.PurePythonEval):
+        _OPS = PKDict({
```

Comment:
This is a fraction of all of the operators defined in defns.rpn. This was enough to cover all test cases and all of the example elegant files you sent. I'm inclined to go through and implement more of the simple defns. Thoughts?

Reply:
Added more defns. I think it is enough for now. https://github.com/radiasoft/sirepo/pull/3180/commits/367de305b57cca4e8a999cfb1cbdce77410755ac